### PR TITLE
WIP: Bounty 006 - Add JSON RPC api V1

### PR DESCRIPTION
**Pull request by troggy**
- [x] fix: crashes on any state change
- [x] add: Websocket transport (socket.io ?)
- [x] add `eth_getBlockByNumber` (so that we can implement block explorer)
- [ ] API docs

----
https://github.com/parsec-labs/parsec-node/issues/10

### CODE REVIEW

File: src/jsonrpc.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      hash,
+      parentHash: block.parent,
+      size: `0x${block.hex().length.toString(16)}`,
+      timestamp: `0x${block.timestamp.toString(16)}`,
+      transactions: txs,
+      uncles: [],
+    };
+  };
+
+  const getBlockByNumber = async (heightOrTag, ...args) => {
```
Comment (Koroqe) [inline src/jsonrpc.js]: In specs it supposed to get exact second parameter - boolean, isn't it?
https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbynumber

File: src/api/jsonrpc.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      hash,
+      parentHash: block.parent,
+      size: `0x${block.hex().length.toString(16)}`,
+      timestamp: `0x${block.timestamp.toString(16)}`,
+      transactions: txs,
+      uncles: [],
+    };
+  };
+
+  const getBlockByNumber = async (heightOrTag, ...args) => {
```
Comment (Koroqe) [inline src/api/jsonrpc.js]: In specs it supposed to get exact second parameter - boolean, isn't it? 
https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbynumber

Comment (troggy) [reply src/api/jsonrpc.js]: yeah, forgot to change this after refactoring for WS. Good catch! 👍

**Comment by troggy** [comment]
rebased

**Comment by johannbarbie** [comment]
rebased

**Comment by sunify** [comment]
WebSocket connection doesn't seem to work. You can test it with the local copy of Alice

**Comment by troggy** [comment]
rebased

**Pull request merged**